rest-service/manager_rest/rest/rest_utils.py

```diff
@@ -644,57 +644,64 @@ def find_recursive_deployments(self, source_id):
         propagate that increase to all deployment that are referenced
         directly and indirectly by `target` deployment
-        :param target: Target Deployment node that we need to attach source to
-        :rtype Deployment
+        :param target_id: Target Deployment ID that we need to attach
+        source to
+        :rtype str
         :param source: Source deployment that reference target deployment
         :rtype Deployment
         """
-        total_services = source.sub_services_count or 0
-        total_environments = source.sub_environments_count or 0
+        total_services = source.sub_services_count
+        total_environments = source.sub_environments_count
         if source.is_environment:
-            total_environments += 1
+            total_environments = total_environments + 1 if \
+                total_environments else 1
         else:
-            total_services += 1
-        target_group = [target.id] + self.find_recursive_deployments(target.id)
+            total_services = total_services + 1 if total_services else 1
+        target_group = [target_id] + self.find_recursive_deployments(target_id)
         for target_id in target_group:
-            target = self.sm.get(
-                models.Deployment,
-                target_id,
-                locking=True
-            )
-            if not target.sub_services_count:
-                target.sub_services_count = total_services
-            else:
-                target.sub_services_count += total_services
-            if not target.sub_environments_count:
-                target.sub_environments_count = total_environments
-            else:
-                target.sub_environments_count += total_environments
-            self.sm.update(target)
+            if total_services or total_environments:
+                target = self.sm.get(
+                    models.Deployment,
+                    target_id,
+                    locking=True
+                )
+                if not target.sub_services_count:
+                    target.sub_services_count = total_services
+                else:
+                    target.sub_services_count += total_services or 0
+                if not target.sub_environments_count:
+                    target.sub_environments_count = total_environments
+                else:
+                    target.sub_environments_count += total_environments or 0
+                self.sm.update(target)
 
-    def decrease_deployment_counts_in_graph(self, source):
+    def decrease_deployment_counts_in_graph(self, target_id, source):
         """
         Decrease the counter for each deployment that is referenced
         directly and indirectly by `source` deployment and the counts that
         need to be update are possibly for `sub_environments_count`
         & `sub_services_count`
+        :param target_id: Target Deployment ID that we need to de-attach
```

Comment:
I think it'd be better to go with the approach of passing the deployment here (as noted above).

Reply:
Answered above, the same reason for locking deployment